### ISSUE TITLE
[JNI] Error while creating Estimator on linux-cpu/gpu

### ISSUE THREAD

**Issue by mandar2812** [eaplatanios/tensorflow_scala#114]
So when running the [cifar.sc](https://github.com/transcendent-ai-labs/DynaML/blob/tf-0.2.0/scripts/cifar.sc) script in DynaML branch _tf-0.2.0_, I get the following stack trace.

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f022fdcb30f, pid=23298, tid=0x00007f02a5c4f700
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b10) (build 1.8.0_171-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtensorflow_jni.so+0x7a30f]  tensorflow::ExtendSessionGraphHelper(TF_Session*, TF_Status*)+0x41f
#
# Core dump written. Default location: /ufs/chandork/Development/DynaML/core or core.23298
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#


Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.platanios.tensorflow.jni.Session$.run(J[B[J[J[I[J[I[JZ[J)[B+0
j  org.platanios.tensorflow.api.core.client.Session.runHelper(Lorg/platanios/tensorflow/api/core/client/FeedMap;Ljava/lang/Object;Ljava/lang/Object;Lscala/Option;ZLorg/platanios/tensorflow/api/core/client/Executable;Lorg/platanios/tensorflow/api/core/client/Fetchable;)Lscala/Tuple2;+547
j  org.platanios.tensorflow.api.core.client.Session.run(Lorg/platanios/tensorflow/api/core/client/FeedMap;Ljava/lang/Object;Ljava/lang/Object;Lscala/Option;Lorg/platanios/tensorflow/api/core/client/Executable;Lorg/platanios/tensorflow/api/core/client/Fetchable;)Ljava/lang/Object;+14
j  org.platanios.tensorflow.api.learn.SessionManager$$anonfun$prepareSession$1.apply(Lorg/platanios/tensorflow/api/ops/Op;)Lscala/collection/Seq;+59
j  org.platanios.tensorflow.api.learn.SessionManager$$anonfun$prepareSession$1.apply(Ljava/lang/Object;)Ljava/lang/Object;+5
J 7563 C2 scala.Option.foreach(Lscala/Function1;)V (19 bytes) @ 0x00007f028d582e34 [0x00007f028d582a60+0x3d4]
j  org.platanios.tensorflow.api.learn.SessionManager.prepareSession(Ljava/lang/String;Lscala/Option;Lscala/Option;ZILscala/Option;Lscala/Option;Lorg/platanios/tensorflow/api/core/client/FeedMap;Lscala/Option;Lscala/Option;)Lorg/platanios/tensorflow/api/core/client/Session;+131
j  org.platanios.tensorflow.api.learn.ChiefSessionCreator.createSession()Lorg/platanios/tensorflow/api/core/client/Session;+201
j  org.platanios.tensorflow.api.learn.HookedSessionCreator.createSession()Lorg/platanios/tensorflow/api/core/client/Session;+8
j  org.platanios.tensorflow.api.learn.RecoverableSession$$anonfun$createSession$1.apply$mcV$sp()V+8
j  org.platanios.tensorflow.api.learn.RecoverableSession$$anonfun$createSession$1.apply()V+1
j  org.platanios.tensorflow.api.learn.RecoverableSession$$anonfun$createSession$1.apply()Ljava/lang/Object;+1
J 11563 C1 scala.util.control.Exception$Catch.apply(Lscala/Function0;)Ljava/lang/Object; (143 bytes) @ 0x00007f028ed6016c [0x00007f028ed60060+0x10c]
j  org.platanios.tensorflow.api.learn.RecoverableSession$.createSession(Lorg/platanios/tensorflow/api/learn/SessionCreator;)Lorg/platanios/tensorflow/api/core/client/Session;+54
j  org.platanios.tensorflow.api.learn.RecoverableSession.<init>(Lorg/platanios/tensorflow/api/learn/SessionCreator;)V+10
j  org.platanios.tensorflow.api.learn.RecoverableSession$.apply(Lorg/platanios/tensorflow/api/learn/SessionCreator;)Lorg/platanios/tensorflow/api/learn/RecoverableSession;+5
j  org.platanios.tensorflow.api.learn.MonitoredSession$.apply(Lorg/platanios/tensorflow/api/learn/SessionCreator;Lscala/collection/immutable/Set;Z)Lorg/platanios/tensorflow/api/learn/MonitoredSession;+55
j  org.platanios.tensorflow.api.learn.estimators.Estimator$.monitoredTrainingSession(Lorg/platanios/tensorflow/api/learn/Configuration;Lscala/collection/immutable/Set;Lscala/collection/immutable/Set;Lorg/platanios/tensorflow/api/learn/SessionScaffold;)Lorg/platanios/tensorflow/api/learn/MonitoredSession;+88
 
```

## Problematic Code

The problem appears when creating an Estimator in the following manner.

```scala
val (model, estimator) = tf.createWith(graph = Graph()) {
        val model = tf.learn.Model.supervised(
          input, architecture,
          target, processTarget,
          loss, optimizer)

        println("\nTraining the regression model.\n")

        val estimator = if(inMemory) {
          tf.learn.InMemoryEstimator(
            model,
            tf.learn.Configuration(Some(summariesDir)),
            stopCriteria,
            Set(
              tf.learn.StepRateLogger(
                log = false, summaryDir = summariesDir,
                trigger = tf.learn.StepHookTrigger(stepRateFreq)),
              tf.learn.SummarySaver(summariesDir, tf.learn.StepHookTrigger(summarySaveFreq)),
              tf.learn.CheckpointSaver(summariesDir, tf.learn.StepHookTrigger(checkPointFreq))),
            tensorBoardConfig = tf.learn.TensorBoardConfig(summariesDir, reloadInterval = checkPointFreq))
        } else {
          tf.learn.FileBasedEstimator(
            model,
            tf.learn.Configuration(Some(summariesDir)),
            stopCriteria,
            Set(
              tf.learn.StepRateLogger(
                log = false, summaryDir = summariesDir,
                trigger = tf.learn.StepHookTrigger(stepRateFreq)),
              tf.learn.SummarySaver(summariesDir, tf.learn.StepHookTrigger(summarySaveFreq)),
              tf.learn.CheckpointSaver(summariesDir, tf.learn.StepHookTrigger(checkPointFreq))),
            tensorBoardConfig = tf.learn.TensorBoardConfig(summariesDir, reloadInterval = checkPointFreq))
        }

        estimator.train(() => training_data)

        (model, estimator)
      }
```

This problem does **not** happen when I run this script on a Mac. I am using tf-scala 0.2.2.

**Comment by mandar2812** [comment]
@eaplatanios So this was seeming like a redux of #112. But its happening at a different part of the code, and not happening on Mac. If you cannot reproduce this on linux-cpu, I'll take a look at my linux setup again.

**Comment by eaplatanios** [comment]
@mandar2812 This is actually the same error as the one I'm getting on the CI server, but not on my Linux machines. I'll investigate this soon. In the meantime, could you please tell me if things work fine for you if you compile the TF binaries yourself from the r1.9 branch? Thanks! :)

**Comment by mandar2812** [comment]
@eaplatanios Do you want me to take the latest commit on the  r1.9 branch, or a particular tag/revision/commit?

**Comment by eaplatanios** [comment]
I think the latest commit in r1.9 works fine for me so that should do it :)

**Comment by evan-wehi** [comment]
I'm getting what appears to be the same thing. Crash log attached. I might wait to see what @mandar2812 comes up with - for some reason I can never get TF to build, even on a fresh ubuntu.

[hs_err_pid34538.log](https://github.com/eaplatanios/tensorflow_scala/files/2126255/hs_err_pid34538.log)

**Comment by eaplatanios** [comment]
@mandar2812 @evan-wehi This has now been fixed in version 0.2.3. :)
